dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -216,86 +220,110 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
             {
+                {
+                    auto lock = s_activatorInfoLock.lock_exclusive();
+                    THROW_HR_IF_MSG(E_INVALIDARG, s_comActivatorRegistration, "ComActivator already registered.");
+                }
+
                 GetWaitHandleForArgs().create();
 
                 THROW_IF_FAILED(::CoRegisterClassObject(
                     taskClsid,
                     winrt::make<PushNotificationBackgroundTaskFactory>().get(),
                     CLSCTX_LOCAL_SERVER,
                     REGCLS_MULTIPLEUSE,
-                    &cookie));
+                    &s_comActivatorRegistration));
             }
 
             if (builder)
             {
                 registeredTaskFromBuilder = builder.Register();
             }
 
-            PushNotificationRegistrationToken token = { cookie, registeredTaskFromBuilder };
             scopeExitToCleanRegistrations.release();
 
-            PushNotificationTelemetry::ActivatorRegisteredByApi(S_OK, details.Options());
+            auto lock = s_activatorInfoLock.lock_exclusive();
+            s_pushTriggerRegistration = registeredTaskFromBuilder;
 
-            return token;
+            PushNotificationTelemetry::ActivatorRegisteredByApi(S_OK, details.Activators());
         }
 
-        catch (...)
+        catch(...)
         {
             HRESULT hrError = wil::ResultFromCaughtException();
-            PushNotificationTelemetry::ActivatorRegisteredByApi(hrError,
-                details == nullptr ? PushNotificationRegistrationOptions::Undefined : details.Options());
+            PushNotificationTelemetry::ActivatorRegisteredByApi(hrError, 
+                details == nullptr ? PushNotificationRegistrationActivators::Undefined : details.Activators());
 
             THROW_HR(hrError);
         }
     }
 
-    void PushNotificationManager::UnregisterActivator(PushNotificationRegistrationToken const& token, PushNotificationRegistrationOptions const& options)
+    void PushNotificationManager::UnregisterActivator(PushNotificationRegistrationActivators const& activators)
     {
         try
         {
-            THROW_HR_IF_NULL(E_INVALIDARG, token);
-            if (WI_IsFlagSet(options, PushNotificationRegistrationOptions::PushTrigger))
+            auto lock = s_activatorInfoLock.lock_exclusive();
+            if (WI_IsFlagSet(activators, PushNotificationRegistrationActivators::PushTrigger))
             {
-                auto taskRegistration = token.TaskRegistration();
-                THROW_HR_IF_NULL(HRESULT_FROM_WIN32(ERROR_NOT_FOUND), taskRegistration);
-                taskRegistration.Unregister(true);
+                THROW_HR_IF_NULL_MSG(HRESULT_FROM_WIN32(ERROR_NOT_FOUND), s_pushTriggerRegistration, "PushTrigger not registered.");
+                s_pushTriggerRegistration.Unregister(true);
+                s_pushTriggerRegistration = nullptr;
```

Comment:
So push triggers are one-registration-per-process? Should this be refcounted somehow?

Reply:
PushTriggers are one-registration-per-process. Also, this is a static winrt object, does it need ref counting?